include/vast/Translation/CodeGenVisitor.hpp

```diff
@@ -40,13 +41,13 @@ namespace vast::cg
     //
     // `MetaGenerator` takes care of attaching location metadata to generated mlir primitives.
     //
-    template< template< typename > class CodeGenVisitorMixin, MetaGeneratorLike MetaGenerator >
+    template< typename CodeGenContext, template< typename > class CodeGenVisitorMixin, MetaGeneratorLike MetaGenerator >
```

Comment:
Umaybe just use `Context` or `CGContext` not to shadow the class `CodeGenContext`.

Reply:
Similarly in other templates.